### ISSUE TITLE
callback or end event is broken (is fired too soon) when downloading

### ISSUE THREAD

**Issue by timaschew** [suiteplus/nscabinet#10]
If you just use it this way

``` js
gulp.task('download', function () {
  nscabinet.download(['foo.bar'], {
    rootPath : SUITE_SCRIPTS
  })
  .pipe(gulp.dest(process.cwd()))
})
```

it's working fine, but if you do some other operations in the gulp pipeline you'll realize that the pipeline continues before the download is finished, because the console log output comes later than the callback or end event.

``` js
gulp.task('download-broken', function () {
  nscabinet.download(['foo.bar'], {
    rootPath : SUITE_SCRIPTS
  })
  .pipe(require('gulp-prompt').confirm('log output of download above comes after me'))
  .pipe(gulp.dest(process.cwd()))
})
```

**Comment by wkrueger** [comment]
I've made some review on the code that may have fixed this, also removing the redundant `event-stream` lib (we had 2 libs for the same thing...).
Also adding tests for this possible kind of issue.
I actually rarely had that kind of use until now, since most of the times I uses nscabinet to upload .js files paired with `gulp.watch`.
Fix will be up for v0.5.x in a couple days.

**Comment by wkrueger** [comment]
This issue is already fixed/patched on the live version and in the incoming one.
